### PR TITLE
Feature/exception handling aspects

### PR DESCRIPTION
**Pull request by wfschneider-rfid**
### What changes were proposed in this pull request?

1. Add additional information to the `banner.txt` file to print the Spring Boot version and the application name and version at startup.
2. Adds a `SmartCosmosException` as a base exception type.
3. Adds a new annotation `@SmartCosmosService` to be used for 
4. Adds an aspect that wraps all public methods in any class that has the `@SmartCosmosService` annotation.  This aspect captures all exceptions, logs information about the exception and turns them into `SmartCosmosException`s.  This ensures all exceptions are caught close to their point of origin so contextual information can be logged to help in the PI/PD process and provides for a consistent exception handling approach.

### How is this patch documented?

Javadoc on affected classes

### How was this patch tested?

Uni tests and postman driven tests to create exception situations

#### Depends On

Nothing but there will be more to follow

### CODE REVIEW

File: smartcosmos-framework/src/main/java/net/smartcosmos/annotation/SmartCosmosService.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package net.smartcosmos.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+import org.springframework.stereotype.Service;
+
+/**
+ * Annotation use with AOP to indicate the annotated element is capavble of participating in AOP based exception handling.
```
Comment (andreassiegelrfid) [inline smartcosmos-framework/src/main/java/net/smartcosmos/annotation/SmartCosmosService.java]: Typo. 😉

File: smartcosmos-framework/src/main/java/net/smartcosmos/aspects/SmartCosmosExtensionAspects.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.stereotype.Component;
+
+import net.smartcosmos.exceptions.SmartCosmosException;
+
+/**
+ * Aspects to be applied to extension services implemented in the SMART COSMOS Objects architecture.
+ * <p>
+ * To use this aspect you will need to enable aspects with @EnableAspectJAutoProxy in your @Configuration.  Then you need to add
```
Comment (andreassiegelrfid) [inline smartcosmos-framework/src/main/java/net/smartcosmos/aspects/SmartCosmosExtensionAspects.java]: I wonder if we can tackle this using custom auto-configuration, like _If there is any class annotated with `@SmartCosmosService`, enable aspects._

File: smartcosmos-framework/src/main/resources/banner.txt [hunk 3]
```diff
@@ -25,3 +25,5 @@
    ;;     ;;`    ;    `;: ;;   ;;   ;          ;;`    ;;.    .;   `;.   ;:      ''''''+
     ;;;;;;;      ;;;;;;,   ;;;;;    ;;;;;;;     :;;;;;;      .;    ,;;;;;        ''''''
 
+${application.title}: ${application.formatted-version}
+Spring Boot Version: ${spring-boot.formatted-version}
```
Comment (andreassiegelrfid) [inline smartcosmos-framework/src/main/resources/banner.txt]: ❤️

File: smartcosmos-framework/src/main/java/net/smartcosmos/annotation/SmartCosmosService.java (hunk 1 above)
Comment (wfschneider-rfid) [inline smartcosmos-framework/src/main/java/net/smartcosmos/annotation/SmartCosmosService.java]: doh!

File: smartcosmos-framework/src/main/java/net/smartcosmos/annotation/SmartCosmosService.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ */
+@Service
+@Target({ ElementType.METHOD, ElementType.TYPE })
```
Comment (wfschneider-rfid) [inline smartcosmos-framework/src/main/java/net/smartcosmos/annotation/SmartCosmosService.java]: Should be `Type` only.  I will fix

File: smartcosmos-framework/src/main/java/net/smartcosmos/aspects/SmartCosmosExtensionAspects.java (hunk 2 above)
Comment (wfschneider-rfid) [inline smartcosmos-framework/src/main/java/net/smartcosmos/aspects/SmartCosmosExtensionAspects.java]: probably.  I did not think of that

Comment (wfschneider-rfid) [inline smartcosmos-framework/src/main/java/net/smartcosmos/aspects/SmartCosmosExtensionAspects.java]: it looks like Spring Boot [already thought of that](https://github.com/spring-projects/spring-boot/blob/v1.4.1.RELEASE/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/aop/AopAutoConfiguration.java)

**Pull request merged**